### PR TITLE
chore: enhance PersiaBatch interface robustness

### PR DESCRIPTION
**Pull request by williamstar**

### CODE REVIEW

File: persia/embedding/data.py [hunk 1]
```diff
@@ -9,11 +11,140 @@
 _logger = get_default_logger()
 
 
-IDTypeFeatureSparse = NewType("IDTypeFeatureSparse", Tuple[str, List[np.ndarray]])
+# Maximum batch_size support.
```
Comment (NOBLES5E) [inline persia/embedding/data.py:14]: maximum batch size supported

File: rust/persia-core/src/data.rs [hunk 2]
```diff
@@ -168,40 +173,84 @@ impl PersiaBatch {
...
+
+        self.id_type_features
+            .as_mut()
+            .unwrap()
+            .push(FeatureBatch::new(id_type_feature_name, indices))
+    }
+
+    /// Set requires_grad for [`PersiaBatchImpl`] and check it
+    /// valid or not. Finally convert the IdTypeFeatureBatch to EmbeddingTensor.
+    pub fn check_batch(&mut self, requires_grad: Option<bool>) -> PyResult<()> {
```
Comment (snowpeakz) [inline rust/persia-core/src/data.rs:223]: `check_batch` sounds like a func that do not modify the struct, how about name it as `consume_batch`?

Comment (karoka) [reply rust/persia-core/src/data.rs:223]: sounds good to me

**Comment by snowpeakz** [comment]
btw, it is better to merge main branch to ensure all of CI procedures fine.

File: rust/persia-core/src/data.rs (hunk 2 above)
Comment (williamstar) [reply rust/persia-core/src/data.rs:223]: `consume_batch` sounds like taken entire `PersiaBatchImpl`, but indeed the function is converted `id_type_feature_batches` to `EmbeddingTensor`, so I think `convert_id_type_features` is better.

Comment (snowpeakz) [reply rust/persia-core/src/data.rs:223]: Sure, how about `convert_to_embedding_tensor`

Comment (williamstar) [reply rust/persia-core/src/data.rs]: After discussion, it is better to modify `check_batch` to `converted_id_type_features2embedding_tensor`.

**Review by snowpeakz — APPROVED** [review]

**Review by NOBLES5E — APPROVED** [review]

**Pull request merged**